cpp/src/arrow/acero/aggregate_node_test.cc

```diff
@@ -210,5 +210,36 @@ TEST(GroupByNode, NoSkipNulls) {
   AssertExecBatchesEqualIgnoringOrder(out_schema, {expected_batch}, out_batches.batches);
 }
 
+TEST(ScalarAggregateNode, AnyAll) {
+  // GH-43768: boolean_any and boolean_all with constant input should work well
+  // when min_count != 0.
+  std::shared_ptr<Schema> in_schema = schema({field("not_used", int32())});
+  std::shared_ptr<Schema> out_schema = schema({field("agg_out", boolean())});
+  std::vector<ExecBatch> batches{
+      ExecBatchFromJSON({int32()}, "[[42], [42], [42], [42]]")};
```

Comment:
Is it possible to test more cases? Something like:
```c++
  struct AnyAllCase {
    std::string batches_json;
    std::string expected_json;
  };
  std::vector<AnyAllCase> cases {
    {"[[42], []]", "[[false]]"},
    {"[[42, 42], []]", "[[true]]"},
    {"[[42], [42]]", "[[true]]"},
    {"[[42], [42, 42]]", "[[true]]"},
    {"[[42], []]", "[[false]]"},
    {"[[42, 42], []]", "[[true]]"},
    {"[[42], [42]]", "[[true]]"},
    {"[[42], [42, 42]]", "[[true]]"},
  };
```

Also vary the `skip_nulls` and the boolean literal.

Reply:
It's a bit hard to build test like this since the Literal column would only built by `Projection` here, but I've tried

Reply:
@pitrou Would you mind take a quick look that whether the testing is ok?